### PR TITLE
[webgl2 spec] "shader-output-incompatible" draw buffer validation.

### LINKED ISSUE #3623: Clarify draw calls with incompatible render targets
The WebGL 2.0 spec says the following:
> If the values written by the fragment shader do not match the format(s) of the corresponding color buffer(s) (for example, the output variable is an integer, but the corresponding color buffer has a floating-point format, or vice versa), and the color buffer states are not set to `NONE` by `DrawBuffers`, draws generate an `INVALID_OPERATION` error; if the color buffer states are set to `NONE` by `DrawBuffers`, they remain untouched.

> If any draw buffer with an attachment does not have a defined fragment shader output, draws generate `INVALID_OPERATION`, unless all 4 channels of `colorMask` are set to false.

1. The language above does not explicitly address the following case: a draw buffer is enabled, has an attachment of an incompatible format (e.g., an integer attachment for a floating-point fragment shader output), and all 4 channels of `colorMask` are set to false.
2. Are there any interactions with `OES_draw_buffers_indexed`? I.e., should indexed color masks be considered for these rules?

### PR DESCRIPTION
**Pull request by kdashg**
Fixes #3623.

### CODE REVIEW

**Review by kenrussell — APPROVED** [review]
Thank you for putting this together; looks good to me.

@lexaknyazev do you have any feedback on this spec update?

File: specs/latest/2.0/index.html [hunk 1]
```diff
@@ -3720,16 +3720,26 @@ <h3>Draw Buffers</h3>
...
+            (For example, the output variable is an integer, but the corresponding color buffer has a floating-point format, or vice versa)
+          </li>
+          <li>OR there is no defined fragment shader output for its location.</li>
+        </ul>
     </p>
 
     <p>
-        If any draw buffer with an attachment does not have a defined fragment shader output, draws
-        generate <code>INVALID_OPERATION</code>, unless all 4 channels of <code>colorMask</code> are set to false.
+        If any draw buffers are "shader-output-incompatible", generate <code>INVALID_OPERATION</code> if:
```
Comment (lexaknyazev) [inline specs/latest/2.0/index.html:3733]: s/generate/draws generate

**Review by lexaknyazev — APPROVED** [review]
LGTM with a trivial nit.

File: specs/latest/2.0/index.html (hunk 1 above)
Comment (kdashg) [inline specs/latest/2.0/index.html:3733]:
```suggestion
        If any draw buffers are "shader-output-incompatible", draws generate <code>INVALID_OPERATION</code> if:
```

**Pull request merged**